app/lib/actions/katello/host/attach_subscriptions.rb

```diff
@@ -0,0 +1,38 @@
+module Actions
+  module Katello
+    module Host
+      class AttachSubscriptions < Actions::Base
+        middleware.use Actions::Middleware::KeepCurrentUser
+
+        def plan(host, pools_with_quantities)
+          sequence do
+            pool_ids = []
+            pools_with_quantities.each do |pool_with_quantities|
+              pool_ids << pool_with_quantities.pool.id
+              pool_with_quantities.quantities.each do |quantity|
+                plan_action(::Actions::Candlepin::Consumer::AttachSubscription, :uuid => host.subscription_facet.uuid,
+                            :pool_uuid => pool_with_quantities.pool.cp_id, :quantity => quantity)
+              end
+            end
+
+            plan_self(:pool_ids => pool_ids, :host_name => host.name)
+          end
+        end
+
+        def finalize
+          ::Katello::Pool.where(:id => input[:pool_ids]).each do |pool|
```

Comment:
not sure what you mean.  It could be used and would be functionality equivalent, but we don't really care about the return value.